### ISSUE TITLE
JSDOM sets the default value of window.onhashchange to undefined which breaks feature detection.

### ISSUE THREAD

**Issue by b01** [tmpvar/jsdom#1354]
Not possible to use feature detection of onhashchange in jsdom 7.2.1 like so:

``` javascript
if ("onhashchange" in window) {
    console.log("The browser supports the hashchange event!");
}
```

The message does not print from the code above. However, this is my work-a-round for now:

```
// HACK: When in a browser, the default value for onhashchange is null, but jsdom sets it to undefined.
window.onhashchange = null;
if ("onhashchange" in window) {
    console.log("The browser supports the hashchange event!");
}
```

**Comment by domenic** [comment]
This should be fixed in master, will test and get back to you...

**Comment by domenic** [comment]
@Sebmaster this looks to be not fixed and also the current situation is confusing. `WindowEventHandlers` does not have an implementation at all, so won't that break the onxyzs on body? It also isn't mixed in to `Window`. (Nor is GlobalEventHandlers.)

A quick fix might be to comment out all of the contents of WindowEventHandlers.idl and GlobalEventHandlers.idl as TODOs. What do you think?

**Comment by Sebmaster** [comment]
To clarify in this issue: There is no change to this in master and this should behave as it did in 7.2.x. To solve this we'll need to properly implement inline event handlers. It's a change that will come soon, but probably not before the next major release.

**Comment by Sebmaster** [comment]
Fixed in 5f3cbd6.
